config/software/openssl.rb

```diff
@@ -119,7 +126,14 @@
   if version.satisfies?("< 3.0.0")
     configure_args += ["--with-fipsdir=#{install_dir}/embedded", "fips"] if fips_mode?
   else
-    configure_args += ["enable-fips"] if fips_mode?
+    # Enable FIPS for OpenSSL 3.x
```

Comment:
I wonder if we could simplify this by always building the fips module for any linux if we're > 3.0? 

fips-mode made sense in the build prior to 3.0, because it was baked into the openssl binary.  Now that the provider mode exists in 3+, it seems like we should be able to always ship the provider.